ci/infra/meta.yml

```diff
@@ -0,0 +1,458 @@
+          AWS_DEFAULT_REGION: ((dataworks.aws_region))
+          ASSUME_DURATION: 7200
+        run:
+          path: sh
+          args:
+            - -exc
+            - |
+              export AWS_DEFAULT_REGION
+              export ASSUME_DURATION
+              source /assume-role
+              set +x
+              ASS_RUN_ARN=$(cat assessment_run_arn/arn)
+              ASS_RUN_DURATION=$(cat terraform-output-management/outputs.json | jq -r '.hardened_assessment_template.value.duration')
+              STATE="unknown"
+              set +x
+              LIMIT=$((${ASS_RUN_DURATION} + 600))     # loop time limit in seconds, set to assessment run duration plus 10 minutes
+              INTERVAL=60                              # loop interval in seconds
+              TIME_PASSED=0                            # loop counter based on seconds passed
+              while [[ "${STATE}" != 'COMPLETED' ]]; do
+                if [ `expr ${TIME_PASSED} % 300` -eq 0 ]; then       # print a message every 300 seconds (5 minutes)
+                  echo "waiting for the assessment run to complete, passed $((${TIME_PASSED} / 60)) minutes out of $((${LIMIT} / 60))"
+                fi
+                STATE=$(aws inspector describe-assessment-runs --assessment-run-arns ${ASS_RUN_ARN} | jq -r '.assessmentRuns[0].state')
+                if [[ "${STATE}" == 'null' ]]; then
+                  echo "Incorrect assessment run ARN: ${ASS_RUN_ARN}"
+                  exit 1
+                fi
+                if [[ ${TIME_PASSED} -gt ${LIMIT} ]]; then
+                  echo "Maximum wait time of $((${LIMIT} / 60)) minutes reached"
+                  exit 1
+                fi
+                sleep ${INTERVAL}
+                TIME_PASSED=$((${TIME_PASSED}+${INTERVAL}))
+              done
+              set -x
+              SCRIPT_PATH="aws-security-tools/files/inspector"
+              VULNS=$(python3 ${SCRIPT_PATH}/evaluate-assessment-run.py --assessment-run-arn ${ASS_RUN_ARN} --whitelist-file ${SCRIPT_PATH}/${WHITELIST_NAME})
+              rc=$?
+              if [[ $rc -ne 0 -o $VULNS -gt 0 ]]; then
+                exit 1
+              fi
+
+    promote-ami:
+      task: promote-ami
+      config:
+        platform: linux
+        image_resource:
+          type: docker-image
+          source:
+            repository: ((dataworks.docker_awscli_repository))
+            version: ((dataworks.docker_awscli_version))
+            tag: ((dataworks.docker_awscli_version))
+        inputs:
+          - name: source_ami
+        params:
+          AWS_REGION: ((dataworks.aws_region))
+          AWS_ROLE_ARN: arn:aws:iam::((dataworks.aws_management_acc)):role/ci
+          AWS_DEFAULT_REGION: ((dataworks.aws_region))
+        run:
+          path: sh
+          args:
+            - -exc
+            - |
+              export AWS_DEFAULT_REGION
+              source /assume-role
+              set +x
```

Comment:
not sure this should be here, I think it's a mistake that's been copied around too much

Reply:
it uses the aws-cli image, it's the only way for it to get credentials.

Reply:
that's not the case, but it is the misconception that leads to this error being repeated. The assume-role script does a `set +x` at the beginning, then a `set -x` at the end
https://github.com/dwp/docker-awscli/blob/master/assume-role

Reply:
to be clear, I'm saying to remove the `set +x` line only

Reply:
Done.